mungegithub/mungers/approval-handler.go

```diff
@@ -90,62 +95,169 @@ func (h *ApprovalHandler) Munge(obj *github.MungeObject) {
 		return
 	}
 
-	approverSet := sets.String{}
+	approverSet := createApproverSet(comments)
+	//stillNeedsApproval is a set of OWNERS paths associated with the files that have yet to be approved
+	approvedFiles, stillNeedsApproval := h.getApprovalNeededOwnersFiles(files, approverSet)
 
-	// from oldest to latest
-	for i := len(comments) - 1; i >= 0; i-- {
-		c := comments[i]
+	if err := h.updateNotification(obj, approvedFiles, stillNeedsApproval); err != nil {
+		return
+	}
+	if stillNeedsApproval.Len() > 0 {
+		if obj.HasLabel(approvedLabel) {
+			glog.Infof("Canceling the approve label because these owners files %v still require a sign-off", stillNeedsApproval)
+			obj.RemoveLabel(approvedLabel)
+		}
+	} else if !obj.HasLabel(approvedLabel) {
+		obj.AddLabel(approvedLabel)
+	}
+}
 
-		if !mungerutil.IsValidUser(c.User) {
-			continue
+func (h *ApprovalHandler) updateNotification(obj *github.MungeObject, approvedFiles, needsApprovalOwnersFiles sets.String) error {
+	notificationMatcher := mungeComment.MungerNotificationName(approvalNotificationName)
+	comments, err := obj.ListComments()
+	if err != nil {
+		glog.Error("Could not list the comments for PR%v", obj.Issue.Number)
+		return err
+	}
+	latestNotification := mungeComment.FilterComments(comments, notificationMatcher).GetLast()
```

Comment:
Doesn't look like me, somebody must have forced me to do that during my sleep :-)